sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -480,7 +480,7 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     try {
       Option(hive.getFunction(db, name)).map(fromHiveFunction)
     } catch {
-      case CausedBy(ex: NoSuchObjectException) if ex.getMessage.contains(name) =>
+      case CausedBy(ex: Exception) if ex.getMessage.contains(s"$name does not exist") =>
```

Comment:
this won't actually work because it'll find the first exception it sees and tries to match the message. You'll need to do this recursively and match all the messages in the exception stack